backend/src/main/java/com/bakdata/conquery/apiv1/query/concept/specific/temporal/CQAbstractTemporalQuery.java

```diff
@@ -26,7 +32,29 @@
 	 * The query being executed, compared to index. Events in preceding will be cut-off to be always before index, or at the same day, depending on the queries specific implementations.
 	 */
 	protected final CQSampled preceding;
-	
+
+	protected abstract PrecedenceMatcher createMatcher();
+
+	@Override
+	public final QPNode createQueryPlan(QueryPlanContext context, ConceptQueryPlan plan) {
```

Comment:
was ist der Vorteil davon, diese Methode hier zu implementieren ?

Reply:
Davor hatten die Implementierungen von dieser Klasse diese Methode implementiert aber mit sehr viel geteilter Logik. Ich hab die gemeinsame Logik eins nach oben gezogen und den nicht geteilten Teil in createMatcher ausgeteilt, wo dann die Logik der TemporalQuery bestimmt wird. So sparen wir uns die Logik mehrfach zu pflegen und mögliche Fehler zu verhindern. Dazu kommt, dass ich es final gemacht habe, dass man darüber stolpert diese Methode zu implementieren, für den Fall, dass man doch einen sonderfall braucht, und sich erstmal überlegen muss ob das eine gute Idee ist.

Also, wie die 2 Queries zueinander stehen müssen. (eg.: Keine Tage davor, mindestens 2 Tage davor, maximal 2 Tage davor etc)